mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/ReceiveDTokenScreen.tsx

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable @typescript-eslint/consistent-type-assertions */
+import * as Clipboard from 'expo-clipboard'
+import React, { useCallback, useEffect, useState } from 'react'
+import { Share, TouchableOpacity, View } from 'react-native'
+import QRCode from 'react-native-qrcode-svg'
+import { ThemedActivityIndicator, ThemedIcon, ThemedScrollView, ThemedText, ThemedTextBasic, ThemedView } from '@components/themed'
+import { useToast } from 'react-native-toast-notifications'
+import { useThemeContext } from '@shared-contexts/ThemeProvider'
+import { useWalletContext } from '@shared-contexts/WalletContext'
+import { tailwind } from '@tailwind'
+import { translate } from '@translations'
+import { NativeLoggingProps, useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { debounce } from 'lodash'
+import { ButtonGroup } from '../../Dex/components/ButtonGroup'
+import DfiIcon from '@assets/images/dfx_buttons/crypto/DFI_icon.png'
+import BtcIcon from '@assets/images/dfx_buttons/crypto/BTC_icon.png'
+import { PortfolioParamList } from '../PortfolioNavigator'
+import { StackScreenProps } from '@react-navigation/stack'
+import { getAssets, getCryptoRoutes, postCryptoRoute } from '@shared-api/dfx/ApiService'
+import { Blockchain, CryptoRoute } from '@shared-api/dfx/models/CryptoRoute'
+import { BuyType } from '@shared-api/dfx/models/BuyRoute'
+import { Asset } from '@shared-api/dfx/models/Asset'
+import { InfoRow, InfoType } from '@components/InfoRow'
+import { InfoText } from '@components/InfoText'
+import { useDFXAPIContext } from '@shared-contexts/DFXAPIContextProvider'
+// import BtcIconSvg from '@assets/images/dfx_buttons/crypto/Bitcoin_icon.svg'
+import BtcTodBtc from '@assets/images/dfx_buttons/crypto/BTC_to_dBTC.svg'
+
+export async function onShare (address: string, logger: NativeLoggingProps): Promise<void> {
+  try {
+    await Share.share({
+      message: address
+    })
+  } catch (error) {
+    logger.error(error)
+  }
+}
+
+export enum CryptoButtonGroupTabKey {
+  DFI = 'DeFiChain',
+  BTC = 'Bitcoin',
+  ETH = 'Ethereum'
+}
+
+export const MinimumBtcAmount = '0.0005 BTC'
+
+type Props = StackScreenProps<PortfolioParamList, 'ReceiveDTokenScreen'>
+
+export function ReceiveDTokenScreen ({
+  route,
+  navigation
+}: Props): JSX.Element {
+  const logger = useLogger()
+  const { isLight } = useThemeContext()
+  const { address } = useWalletContext()
+  const [activeButton, setActiveButton] = useState<CryptoButtonGroupTabKey>(route.params?.crypto ?? CryptoButtonGroupTabKey.DFI)
+
+  const [bitcoinAddress, setBitcoinAddress] = useState('')
+  const activeAddress = activeButton === CryptoButtonGroupTabKey.DFI ? address : bitcoinAddress
+  const [isLoading, setIsLoading] = useState(false)
+  const defaultFee = 1.2
```

Comment:
fee is already being delivered through the route objects

Reply:
is loaded, just a fallback to make **sure** a fee is shown